customization-base/src/main/java/com/azure/autorest/customization/ClassCustomization.java

```diff
@@ -87,6 +87,51 @@ public ClassCustomization addImports(String... imports) {
         return this;
     }
 
+    /**
+     * Adds a static block to the class. The {@code staticCodeBlock} should include the static keyword followed by
+     * the static code.
+     * @param staticCodeBlock The static code block including the static keyword.
+     * @return The updated {@link ClassCustomization}.
+     */
+    public ClassCustomization addStaticBlock(String staticCodeBlock) {
+        return addStaticBlock(staticCodeBlock, null);
+    }
+
+    /**
+     * Adds a static block to the class. The {@code staticCodeBlock} should include the static keyword followed by
+     * the static code.
+     * @param staticCodeBlock The static code block including the static keyword.
+     * @param importsToAdd The list of imports to add to the class.
+     * @return The updated {@link ClassCustomization}.
+     */
+    public ClassCustomization addStaticBlock(String staticCodeBlock, List<String> importsToAdd) {
+        int staticBlockStartLine;
+        // Find all constructor and field symbols.
+        List<SymbolInformation> staticBlockLocationFinder = languageClient.listDocumentSymbols(fileUri).stream()
+                .filter(symbol -> symbol.getKind() == SymbolKind.FIELD)
+                .collect(Collectors.toList());
```

Comment:
We could change this to only getting the last found field instead of collecting them by using the logic outlined here: https://www.geeksforgeeks.org/find-the-last-element-of-a-stream-in-java/